components/TopNav.vue

```diff
@@ -90,45 +90,78 @@
                         class="hover:text-primary-hover transition-colors"
                     >{{ t('topNav.submit') }}
                     </NuxtLink>
-                    <!-- Profile Section -->
-                    <div
+                    <!-- Profile Picture (if logged in) -->
+                    <button
                         v-if="authStore.isLoggedIn"
                         data-testid="topnav-profile-section"
                         class="flex text-primary"
+                        @click="toggleProfileMenu"
                     >
-                        <!-- Moderation Link -->
-                        <NuxtLink
-                            to="/moderation"
-                            class="hover:text-primary-hover transition-colors text-wrap mr-4"
-                            data-testid="top-nav-mod-link"
-                        >{{
-                            t('topNav.moderation') }}
-                        </NuxtLink>
-                        <!-- Logout Link -->
-                        <NuxtLink
-                            to="/"
-                            class="mr-4"
+                        <img
+                            :src="authStore.userProfileImage"
+                            alt="profile icon"
+                            title="profile icon"
+                            class="w-8 h-8 stroke-primary inline stroke-2 rounded-full mx-1 hover:scale-105"
                         >
-                            <div
-                                class="text-primary"
-                                @click="logout()"
-                            >
-                                {{ t('topNav.logout') }}
-                            </div>
+                    </button>
+                    <!-- Login Button (if logged out)  -->
+                    <div
+                        v-if="!authStore.isLoggedIn"
+                        data-testid="topnav-login"
+                        class="flex text-primary"
+                    >
+                        <NuxtLink to="/login">
+                            {{ t('topNav.login') }}
                         </NuxtLink>
-                        <div class="flex">
-                            <!-- Profile Icon -->
+                    </div>
+                    <!-- Profile Dropdown Menu Options (if profile pic clicked) -->
+                    <div
+                        v-if="profileMenuIsOpen"
+                        v-close-on-outside-click="toggleProfileMenu"
+                        class="absolute border border-black/10 w-[8rem] top-20 right-[5.5rem]
+                        z-10 bg-white rounded-xl p-2 shadow-xl"
```

Comment:
avoid using a hard coded color `white` here as well. This will break dark mode. You can change to the dark theme to test